.github/workflows/publish.yml

```diff
@@ -14,7 +14,9 @@ jobs:
     steps:
       - name: Checkout Code
         uses: actions/checkout@v2
-
+      - name: Get release version
+        id: get_version
+        run: echo ::set-env name=RELEASE_VERSIONS::$(echo "${GITHUB_REF}" | sed -e "s/refs\/tags\///g" | sed -E "s/v?([0-9]+)\.([0-9]+)\.([0-9]+)(-[a-zA-Z]+(\.[0-9]+)?)?/\1.\2.\3\4,\1.\2\4,\1\4/g")
```

Comment:
I'm not 100% sure about this `sed`, it's the sort of stuff that if I were to write in regular code, I'd write a lot of unit tests for, and the consequence of making a mistake here is publishing something to the wrong version and potentially affecting people's actual running machines.

E.g.
```
$ echo "v1.2.3-alpha.3" | sed -E "s/v?([0-9]+)\.([0-9]+)\.([0-9]+)(-[a-zA-Z]+(\.[0-9]+)?)?/\1.\2.\3\4,\1.\2\4,\1\4/g"
1.2.3-alpha.3,1.2-alpha.3,1-alpha.3
```
But then also
```
$ echo "v1.2.4-alpha.3" | sed -E "s/v?([0-9]+)\.([0-9]+)\.([0-9]+)(-[a-zA-Z]+(\.[0-9]+)?)?/\1.\2.\3\4,\1.\2\4,\1\4/g"
1.2.4-alpha.3,1.2-alpha.3,1-alpha.3
```
notice that `1.2-alpha.3,1-alpha.3` is repeated for basically a path version and potentially multiple alphas behind.

Generally though (this is not criticism!) this kind of regex parsing feels a bit too brittle, and is a hotbed for bugs that might cause us to actually release something to a version we don't want to. (I wouldn't trust myself to not make mistakes here, even reviewing this..)

Have you looked into existing Github actions out there that do similar stuff?

Or maybe we can use [this script](https://github.com/cloudflare/semver_bash/blob/master/semver.sh)?

Reply:
We fixed this concern by forking the action repo and addressing it in code, see [Enrico2/Publish-Docker-Github-Action](http://github.com/Enrico2/Publish-Docker-Github-Action)